plugins/irma-client/server-state.js

```diff
@@ -113,13 +123,13 @@ module.exports = class ServerState {
         return;
       }
 
-      fetch(this._options.polling.url(this._options))
-      .then(r => {
-        if ( r.status != 200 )
-          throw(`Error in fetch: endpoint returned status other than 200 OK. Status: ${r.status} ${r.statusText}`);
-        return r;
+      // On Firefox for Android pending fetch request are actively aborted when navigating.
+      // So in case of an error, we do a second attempt to assure the error is permanent.
+      this._pollOnce()
+      .catch(() => {
+        if (this._options.debugging) console.log('Polling attempt failed; doing a second attempt to confirm error');
+        return this._pollOnce();
```

Comment:
Is this necessary for the rest of the MR to work, and if so, how?

Reply:
Yes and no. Strictly it is not part of the fix indeed. However, when I fixed the `intent://` bug started testing on Firefox, I saw that rather often it randomly failed on Firefox. It failed that often that I thought that I had to fix that too. Otherwise, the previous bug fix wouldn't be very useful.